azurerm/internal/services/compute/linux_virtual_machine_scale_set_resource.go

```diff
@@ -433,6 +434,18 @@ func resourceLinuxVirtualMachineScaleSetCreate(d *schema.ResourceData, meta inte
 		virtualMachineProfile.ExtensionProfile.ExtensionsTimeBudget = utils.String(v.(string))
 	}
 
+	// otherwise the service return the error:
+	// Automatic OS Upgrade is not supported for this Virtual Machine Scale Set because a health probe or health extension was not specified.
+	if upgradeMode == compute.Automatic && len(automaticOSUpgradePolicyRaw) > 0 && (healthProbeId == "" && !hasHealthExtension) {
```

Comment:
I don't think we'll get here, line 352 above will kick us out first if `health_probe_id` is unset, so that will need investigating for refactoring.

Reply:
Yeah... something is messed up during the merging from master. I need to check these carefully again.